pkg/plugins/policies/meshretry/api/v1alpha1/validator.go

```diff
@@ -174,6 +178,29 @@ func validateRateLimitedBackOff(rateLimitedBackOff *RateLimitedBackOff) validato
 	return verr
 }
 
+func validateHostSelection(predicates *[]Predicate) validators.ValidationError {
+	var verr validators.ValidationError
+	path := validators.RootedAt("hostSelection")
+
+	for i, predicate := range *predicates {
+		switch predicate.PredicateType {
+		case OmitHostsWithTags:
+			if predicate.Tags == nil {
+				verr.AddViolationAt(path.Index(i).Field("tags"), validators.MustBeDefined)
+			}
+		case OmitPreviousPriorities:
+			if predicate.UpdateFrequency <= 0 {
+				verr.AddViolationAt(path.Index(i).Field("updateFrequency"), validators.MustBeDefinedAndGreaterThanZero)
+			}
+		case OmitPreviousHosts:
+		default:
+			verr.AddViolationAt(path.Index(i).Field("predicate"), fmt.Sprintf("unknown predicate type '%v'",
+				predicate.PredicateType))
+		}
+	}
+	return verr
+}
```

Comment:
would be cool to cover this in `validator_test.go`

Reply:
Added